src/pages/Login.js

```diff
@@ -0,0 +1,163 @@
+import React, { useState } from 'react'
+import styled from 'styled-components/macro'
+import SubmitButton from '../components/Buttons/SubmitButton'
+import AddButton from '../components/Buttons/AddButton'
+import UserHeader from '../components/UserHeader'
+import { Link } from 'react-router-dom'
+import { useForm } from 'react-hook-form'
+
+export default function Login({ logIn, resetPassword, profile, setProfile }) {
+  const { register, handleSubmit, errors, setError } = useForm()
+  const [loginCounter, setLoginCounter] = useState(0)
+  return (
+    <>
+      <UserHeader />
+      <main>
+        <WrapperStyled>
+          <form onSubmit={handleSubmit(onSubmit)} data-cy="login">
+            <div className="email">
+              <input
+                ref={register}
+                type="email"
+                name="email"
+                placeholder="Enter your E-mail"
+                required
+              />
+              {errors.email && errors.email.type === 'notFound' && (
+                <Error>{errors.email.message}</Error>
+              )}
+            </div>
+            <div className="password">
+              <input
+                ref={register}
+                type="password"
+                name="password"
+                placeholder="Password"
+                required
+              />
+              {errors.password && errors.password.type === 'reset' && (
+                <Error>
+                  {errors.password.message}
+                  <span onClick={handleReset}> reset your password.</span>
+                </Error>
+              )}
+              {errors.password && errors.password.type === 'tooMany' && (
+                <Error>{errors.password.message}</Error>
+              )}
+            </div>
+
+            <div className="login">
+              <SubmitButton text="Log in" className="login" />
+            </div>
+          </form>
+          <section>
+            <div>
+              <p>or</p>
+            </div>
+            <div className="signup">
+              <Link to="/signup">
+                <AddButton text="Sign up" />
+              </Link>
+            </div>
+          </section>
+        </WrapperStyled>
+      </main>
+    </>
+  )
+
+  function onSubmit(data) {
+    setProfile(data)
+    logIn(data)
+      .then((res) => {
+        if (res.code === 'auth/user-not-found') {
+          return setError(
+            'email',
+            'notFound',
+            'The email or password you entered did not match our records. Please double-check and try again.'
+          )
+        }
+        if (res.code === 'auth/wrong-password' && loginCounter <= 3) {
+          setLoginCounter(loginCounter + 1)
+          return setError(
+            'password',
+            'reset',
+            'The email or password you entered did not match our records. You may try again or  '
+          )
```

Comment:
Isn't there something missing after the or?

Reply:
the rest of the message is separated as a link to reset the password :)